lib/Folder/FolderManager.php

```diff
@@ -453,10 +453,16 @@ public function getFoldersForGroups(array $groupIds, int $rootStorageId = 0): ar
 				'a',
 				$query->expr()->eq('f.folder_id', 'a.folder_id')
 			)
-			->where($query->expr()->in('a.group_id', $query->createNamedParameter($groupIds, IQueryBuilder::PARAM_STR_ARRAY)));
+			->where($query->expr()->in('a.group_id', $query->createParameter('groupIds')));
 		$this->joinQueryWithFileCache($query, $rootStorageId);
 
-		$result = $query->executeQuery()->fetchAll();
+        // add chunking because Oracle can't deal with more than 1000 values in an expression list for in queries.
+        $result = [];
+        foreach (array_chunk($groupIds, 1000) as $chunk) {
+            $query->setParameter('groupIds', $chunk, IQueryBuilder::PARAM_INT_ARRAY);
```

Comment:
```suggestion
            $query->setParameter('groupIds', $chunk, IQueryBuilder::PARAM_STR_ARRAY);
```

Probably should fix the tests